### PR TITLE
Listen to weave:service:reset-file-log instead of weave:ui:*

### PR DESCRIPTION
**Pull request by eoger**
See [Bug 1439777](https://bugzilla.mozilla.org/show_bug.cgi?id=1439777) for context.
This takes on a double approach:
- From now on, Firefox will send `success-log-written`/`error-log-written` alongside with `weave:service:reset-file-log` in `data`. This allows us to listen to `reset-file-log` directly instead of `weave:ui:*` which is going away.
- For old versions of Firefox not sending that `data` field, we fall back to `isLastLogFileError()`.

### CODE REVIEW

**Comment by mhammond** [comment]
I think we actually want the same semantics as shouldReportError - which is quite different from "was an error log written" - eg, error logs are written on network errors, but shouldReportError will return false.

I think just cloning the "core" logic (ie, ignoring prolonged etc) from shouldReportError into the addon is best.

**Comment by eoger** [comment]
Thanks Mark, I've made the changes.

File: bootstrap.js [hunk 1]
```diff
@@ -172,14 +176,62 @@ function startoverObserver(subject, topic, data) {
...
+  } finally {
+    iterator.close();
+    return false;
+  }
+  return latestLog.type == "error";
+}
+
+async function shouldReportError(data) {
+  if (data == "success-log-written" ||
+      (!data && !(await isLastLogFileError()))) {
```
Comment (mhammond) [inline bootstrap.js]: I don't think we need to look at log files at all for this. ISTM that listening for "weave:service:sync:finish", "weave:service:login:error" and "weave:service:sync:error" and then calling this function to follow the logic below is all we need (although we probably want an additional check for Status.service == STATUS_OK, to handle the sync:finish case.

**Comment by eoger** [comment]
Updated thanks!

**Pull request merged**